code/modules/lighting/lighting_atom.dm

```diff
@@ -112,42 +114,42 @@
 	var/obj/effect/dummy/lighting_obj/moblight/mob_light_obj = new (src, _range, _power, _color, _duration)
 	return mob_light_obj
 
-
-/atom/proc/set_light_range(new_range)
-	if(new_range == light_range)
-		return
-	SEND_SIGNAL(src, COMSIG_ATOM_SET_LIGHT_RANGE, new_range)
-	. = light_range
-	light_range = new_range
-
-
+/// Setter for the light power of this atom.
 /atom/proc/set_light_power(new_power)
 	if(new_power == light_power)
 		return
-	SEND_SIGNAL(src, COMSIG_ATOM_SET_LIGHT_POWER, new_power)
 	. = light_power
 	light_power = new_power
+	SEND_SIGNAL(src, COMSIG_ATOM_SET_LIGHT_POWER, new_power, .)
```

Comment:
Why did you move the signal to afterwards?
We try to have them as soon as we can whenever possible, because that allows us to add the option to cancel it, for example, if needed.

Reply:
Because this is the standard for setter procs with signals as far as I can tell.

Reply:
This is not the standard at all.

Reply:
Done. More overhead, but done.